src/crawlee/static_content_crawler/_static_content_crawler.py

```diff
@@ -0,0 +1,203 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Any, Generic
+
+from pydantic import ValidationError
+from typing_extensions import NotRequired, TypeVar
+
+from crawlee import EnqueueStrategy
+from crawlee._request import BaseRequestData
+from crawlee._utils.docs import docs_group
+from crawlee._utils.urls import convert_to_absolute_url, is_url_absolute
+from crawlee.basic_crawler import BasicCrawler, BasicCrawlerOptions, ContextPipeline
+from crawlee.errors import SessionError
+from crawlee.http_clients import HttpxHttpClient
+from crawlee.static_content_crawler._static_crawling_context import (
+    HttpCrawlingContext,
+    ParsedHttpCrawlingContext,
+    TParseResult,
+)
+
+if TYPE_CHECKING:
+    from collections.abc import AsyncGenerator, Iterable
+
+    from typing_extensions import Unpack
+
+    from crawlee._types import BasicCrawlingContext, EnqueueLinksFunction, EnqueueLinksKwargs
+    from crawlee.static_content_crawler._static_content_parser import StaticContentParser
+
+TCrawlingContext = TypeVar('TCrawlingContext', bound=ParsedHttpCrawlingContext)
+
+
+@docs_group('Data structures')
+class StaticContentCrawlerOptions(Generic[TCrawlingContext], BasicCrawlerOptions[TCrawlingContext]):
+    additional_http_error_status_codes: NotRequired[Iterable[int]]
+    """Additional HTTP status codes to treat as errors, triggering automatic retries when encountered."""
+
+    ignore_http_error_status_codes: NotRequired[Iterable[int]]
+    """HTTP status codes typically considered errors but to be treated as successful responses."""
+
+
+@docs_group('Classes')
+class StaticContentCrawler(Generic[TCrawlingContext, TParseResult], BasicCrawler[TCrawlingContext]):
```

Comment:
We should make it clear this is not meant to be used directly.

- maybe we could put it into another docs group?
- it might make sense to make it an ABC, but there is no abstract method :smile: 
  - perhaps the context pipeline build could be moved into an abstract method?
- if all else fails, we can just write it in the docblock

Reply:
There is this part of text in doc string:
```
...
    StaticContentCrawler is generic class and is expected to be used together with specific parser that will be used to
    parse http response and type of expected TCrawlingContext which is available to the user function.
    See prepared specific version of it: BeautifulSoupCrawler, ParselCrawler or HttpCrawler for example.
...
```
I added ABC and changed it to docs for _Abstract classes_, but that seems to me kind of very cosmetic in this case :-)

Reply:
See? Nobody reads the docs :smile:  Thanks!